api/Clspv.h

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 The Clspv Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CLSPV_API_CLSPV_H_
+#define CLSPV_API_CLSPV_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+typedef enum ClspvError {
+  CLSPV_SUCCESS = 0,
+  CLSPV_INVALID_CONTAINER,
+  CLSPV_INVALID_ARG,
+  CLSPV_OUT_OF_HOST_MEM,
+  CLSPV_ERROR
+} ClspvError;
+
+typedef struct ClspvContainerImpl *ClspvContainer;
+
+#ifdef _WIN32
+#define EXPORT __declspec(dllexport)
+#else
+#define EXPORT __attribute__((visibility("default")))
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Create Clspv container.
+//
+// |container| - Opaque handle to created Clspv container.
+EXPORT ClspvError clspvCreateContainer(ClspvContainer *container);
```

Comment:
My oversight there perhaps 😅 - Will update and push.

Reply:
Actually now I remember... we need to pass container ptr by reference (otherwise its just a copy).

We could modify create function to instead return ptr to created container. That might be a better approach.